qlik-visualizationextensions/index.d.ts

```diff
@@ -0,0 +1,2417 @@
+
+        /**
+         * Checks if a given mode is allowed.
+         * @param {NavigationModeType} mode - Can be one of the strings: edit   |   analysis
+         * or one of the constants: qlik.navigation.EDIT   |   qlik.navigation.ANALYSIS
+         * @return {boolean} - A Boolean value (true or false).
+         */
+        isModeAllowed(mode: NavigationModeType): boolean;
+
+        /**
+         * Go to the next sheet in the current app. It will do nothing if you do not have sheets in the current context.
+         * The method will return before the actual navigation takes place.
+         * @return {NavigationResult} - A navigation result object.
+         */
+        nextSheet(): NavigationResult;
+
+        /**
+         * Jumps to the previous sheet in the current app. It will do nothing if you do not have sheets in the current context.
+         * The method will return before the actual navigation takes place
+         * @return {NavigationResult} - A navigation result object.
+         */
+        prevSheet(): NavigationResult;
+
+        /**
+         * Sets the current working mode of Qlik Sense.
+         * @param {string | constant} mode - Can be one of the strings: edit   |   analysis
+         * or one of the constants: qlik.navigation.EDIT   |   qlik.navigation.ANALYSIS
+         * @return {NavigationResult} - A navigation result object.
+         */
+        setMode(mode: string): NavigationResult;
+
+        /**
+         * Switches the working mode of Qlik Sense.
+         * @param {NavigationModeType} mode - Can be one of the strings: edit   |   analysis
+         * or one of the constants: qlik.navigation.EDIT   |   qlik.navigation.ANALYSIS
+         * @return {NavigationResult} - A navigation result object.
+         */
+        switchMode(mode: NavigationModeType): NavigationResult;
+    }
+}
+
+declare namespace SelectionStateAPI {
+    interface IQFieldSelections {
+        /**
+         * Selection state for a field.
+         */
+        fieldName: string;
+
+        /**
+         * Sort index. Starting from 0.
+         */
+        qSortIndex: number;
+
+        /**
+         * Reference to the field.
+         */
+        field: FieldAPI.IQField;
+
+        /**
+         * Defines if the field is locked.
+         */
+        locked: boolean;
+
+        /**
+         * Defines if the field is numeric.
+         */
+        isNumeric: boolean;
+
+        /**
+         * Total number of values in the field.
+         */
+        totalCount: number;
+
+        /**
+         * Number of selected values.
+         */
+        selectedCount: number;
+
+        /**
+         * Number of values that will be listed.
+         */
+        qSelectionThreshold: number;
+
+        /**
+         * Object with number of values in different states.
+         */
+        qStateCounts: any;
+
+        /**
+         * Concatenated string of selected values if number of values are less than the threshold, or if the string is of fo
```

Comment:
@minestarks sorry I don't understand that comment to that line. Could it be that you mean another line in the code, please give me a hint